uaa/src/test/java/org/cloudfoundry/identity/uaa/mock/providers/IdentityProviderEndpointsAliasMockMvcTests.java

```diff
@@ -327,229 +373,351 @@ protected UpdateBase(final boolean aliasFeatureEnabled) {
+                    assertThat(aliasIdp).isPresent();
+                    assertIdpReferencesOtherIdp(aliasIdp.get(), updatedOriginalIdp);
+                    assertThat(aliasIdp.get().getName()).isNotBlank().isEqualTo(newName);
+
+                    // check if both have the same non-empty relying party secret in the DB
+                    assertIdpAndAliasHaveSameRelyingPartySecretInDb(updatedOriginalIdp);
+
+                    // check if the returned IdP has a redacted relying party secret
+                    assertRelyingPartySecretIsRedacted(updatedOriginalIdp);
+                }
 
-            @Test
-            void shouldReject_AliasNotSupportedForIdpType_UaaToCustomZone() throws Exception {
-                shouldReject_AliasNotSupportedForIdpType(IdentityZone.getUaa(), customZone);
-            }
+                @Test
+                void shouldReject_AliasIdNotSetInPayload_UaaToCustomZone() throws Exception {
+                    shouldReject_AliasIdNotSetInPayload(IdentityZone.getUaa(), customZone);
+                }
 
-            @Test
-            void shouldReject_AliasNotSupportedForIdpType_CustomZone() throws Exception {
-                shouldReject_AliasNotSupportedForIdpType(customZone, IdentityZone.getUaa());
-            }
+                @Test
+                void shouldReject_AliasIdNotSetInPayload_CustomToUaaZone() throws Exception {
+                    shouldReject_AliasIdNotSetInPayload(customZone, IdentityZone.getUaa());
+                }
 
-            private void shouldReject_AliasNotSupportedForIdpType(final IdentityZone zone1, final IdentityZone zone2) throws Exception {
-                final IdentityProvider<?> uaaIdp = buildUaaIdpWithAliasProperties(zone1.getId(), null, null);
-                final IdentityProvider<?> createdProvider = createIdp(zone1, uaaIdp);
-                assertThat(createdProvider.getAliasZid()).isBlank();
+                private void shouldReject_AliasIdNotSetInPayload(
+                        final IdentityZone zone1,
+                        final IdentityZone zone2
+                ) throws Exception {
+                    final IdentityProvider<?> existingIdp = createIdpWithAlias(zone1, zone2);
+
+                    existingIdp.setAliasId(null);
+                    existingIdp.setName("some-new-name");
+                    shouldRejectUpdate(zone1, existingIdp, HttpStatus.UNPROCESSABLE_ENTITY);
+                }
+
+                @Test
+                void shouldAccept_ShouldFixDanglingRefByCreatingNewAlias_UaaToCustomZone() throws Exception {
+                    shouldAccept_ShouldFixDanglingRefByCreatingNewAlias(IdentityZone.getUaa(), customZone);
+                }
+
+                @Test
+                void shouldAccept_ShouldFixDanglingRefByCreatingNewAlias_CustomToUaaZone() throws Exception {
+                    shouldAccept_ShouldFixDanglingRefByCreatingNewAlias(customZone, IdentityZone.getUaa());
+                }
```

Comment:
Looks good. Leaving comments to track my progress as this is challenging to read through.